kuri_edu/src/kuri_edu/safety_controller.py

```diff
@@ -0,0 +1,123 @@
+import threading
+
+import geometry_msgs.msg
+import rospy
+
+import mayfield_msgs.msg
+import mayfield_utils
+import mobile_base
+
+
+class SafetyController(object):
+
+    # A set of the safety events we're equipped to handle.
+    # It's just bumpers so we don't need to deal with compexities
+    # like "What happens if a front bump and a rear cliff triggers
+    # at the same time"
+    HANDLED_EVENTS = {'BPR_bp', 'BPM_bp', 'BPL_bp'}
+    SAFETY_HZ = 10
+
+    def __init__(self):
+
+        self._pub = rospy.Publisher(
+            "node_online",
+            mayfield_msgs.msg.NodeStatus,
+            latch=True,
+            queue_size=1,
+        )
+
+        # Controllers need to be loaded before we can run
+        mayfield_utils.wait_for_nodes(
+            node_names=['controllers'],
+        )
+
+        self._safety_client = mobile_base.SafetyClient()
+        # Override the safety controller for all events that we don't
+        # handle so they'll be ignored by the hardware
+        self.UNHANDLED_EVENTS = (
+            self._safety_client.safety_statuses()  # All statuses
+            - self.HANDLED_EVENTS                  # Minus the ones we handle
+        )
+        self._safety_client.safety_override(self.UNHANDLED_EVENTS)
+
+        self._sync_lock = threading.Lock()
+        self._block_twists = False
+
+        # Set up publishers before subscribers because once subscribers are
+        # set up, we can start getting callbacks
+        self._cmd_vel_pub = rospy.Publisher(
+            "mobile_base/commands/velocity",
+            geometry_msgs.msg.Twist,
+            queue_size=1
+        )
+
+        self._cmd_vel_sub = rospy.Subscriber(
+            "cmd_vel",
+            geometry_msgs.msg.Twist,
+            self._forward_twists
+        )
+
+    def run(self):
+
+        # Indicate to the world that we're running and ready to go:
+        self._pub.publish(
+            mayfield_msgs.msg.NodeStatus("safety_controller", True)
+        )
+
+        rate = rospy.Rate(self.SAFETY_HZ)  # 10hz by default
+        while not rospy.is_shutdown():
+            try:
+
+                current_status = self._safety_client.get_safety_status()
+
+                if self.HANDLED_EVENTS.intersection(current_status) != set():
```

Comment:
Minor: I think you can omit `!= set()`. The empty `set` will evaluate to `False` I believe.

```python
>>> if not set():
...     print "do stuff"
do stuff
```